02-peer-protocol.md

```diff
@@ -797,6 +797,56 @@ A fulfilling node:
   transaction, AND is past this fulfillment deadline:
     - MUST fail the channel.
 
+### Bounding exposure to trimmed in-flight HTLCs: `max_dust_htlc_exposure_msat`
+
+When a HTLC is present in a channel is below the "trimmed" threshold in [BOLT3 #3](03-transactions.md),
+the HTLC is not claimable on-chain, instead being turned into additional miner
+fees if either party unilaterally closes the channel. Because the threshold is
+per-HTLC, the total exposure to such HTLCs may be substantial if there are many
+dust HTLCs present when the channel is force-closed.
+
+This can be exploited in griefing attacks or even in miner-extractable-value attacks,
+if the malicious entity avails <sup>[mining capabilities](https://lists.linuxfoundation.org/pipermail/lightning-dev/2020-May/002714.html)</sup>.
+
+The total exposure is given by the following back-of-the-envelope computation:
+
+	counterparty's `max_accepted_htlcs` * (`HTLC-success-kiloweight` * opener's `feerate_per_kw` + counterparty's `dust_limit_satoshis`)
+		+ holder's `max_accepted_htlcs` * (`HTLC-timeout-kiloweight` * opener's `feerate_per_kw` + counterparty's `dust_limit_satoshis`)
+
+
+To mitigate this scenario, a `max_dust_htlc_exposure_msat` can be apply at
+HTLC sending, forwarding and receiving.
+
+A node:
+  - upon an incoming HTLC:
+    - if a HTLC's `amount_msat` is inferior to the counterparty's `dust_limit_satoshis` plus the HTLC-timeout fee at the `feerate_per_kw`:
```

Comment:
nit: if "the" HTLC's (...)

And same for the other requirements below. Technically we should probably say "a hash-time-lock contract" but the spec uses "an HTLC' everywhere, so we should be consistent.